src/query/parser/promql/parse.go

```diff
@@ -400,7 +442,26 @@ func (p *parseState) walk(node pql.Node) error {
 
 		return p.addLazyUnaryTransform(unaryOp)
 
+	case *pql.SubqueryExpr:
+		opts.FetchRange = n.Range
+		opts.Step = n.Step
+		opts.Offset = opts.Offset + n.Offset
+		opts.StartOffset = opts.StartOffset + n.Range
+		opts.ShouldOverride = true
+
+		if err := p.walk(n.Expr, opts); err != nil {
+			return err
+		}
+
+		return p.addLazySubqueryTransform(n)
```

Comment:
Does this need to happen before/after walking the rest of the tree?  Just curious how it works by applying afterwards.